emoji/src/main/java/com/vanniktech/emoji/emoji/Emoji.java

```diff
@@ -19,24 +19,29 @@
 
   @NonNull private final String unicode;
   @DrawableRes private final int resource;
+  private final boolean isDuplicate;
```

Comment:
Can we store a reference to the one that is replacing this Emoji instead? That way we could trace them back and have a nice linked structure.

Reply:
A reference to the object is hard (since these are added to the array at the same time, so we habe no instance of the other emoji, while instantiating this one), but the unicode would be possible. Would that suffice?

Reply:
I'd be happy with that.

Reply:
I just had the thought that we *may* be able to abuse the variant mechanism for that. Variants are not shown per default in the picker and we could filter the variants to show in the variant popup by the same mechanism, but way more performant since that are only 6ish elements at most.

Semantically it is a bit off, but duplicates could still be considered "variants" I think.

What do you think?

Reply:
Not a bad idea at all!

Reply:
Can't we first construct faceplam emoji 1? And then make emoji 2 contain a reference to emoji1 since it's duplicated?

I would not misuse the `variants`. That's a different principle and we should mix them up.

Reply:
Possible, but hard. We'd need quite some additional logic in the generator and the templates for that.

Reply:
Then let's leave it as is with the boolean

Reply:
Let's to that then. Might come back to it in a follow up.